### PR TITLE
Only ask for more prekeys when we've run out of them

### PR DESCRIPTION
**Pull request by typfel**

### CODE REVIEW

File: Source/Model/UserClient/UserClient.swift [hunk 1]
```diff
@@ -464,15 +464,14 @@ public extension UserClient {
...
             numberOfKeysRemaining -= 1
         }
-        if(numberOfKeysRemaining < 0) { // this will recover from the fact that the number might already be < 0
-                                        // from a previous run
+        if numberOfKeysRemaining < 0 { // this will recover from the fact that the number might already be < 0
+                                       // from a previous run
             numberOfKeysRemaining = 0;
         }
-        if(oldValue != 0) {
+        if numberOfKeysRemaining == 0 {
```
Comment (marcoconti83) [inline Source/Model/UserClient/UserClient.swift]: Is this line was causing the bug? Was it basically always uploading keys every time a new session was created?

Comment (typfel) [inline Source/Model/UserClient/UserClient.swift]: Yes, as far as I can tell we were uploading a 100 new prekeys on every new session.

Comment (marcoconti83) [inline Source/Model/UserClient/UserClient.swift]: Looking at the linked ticket, it had nothing to do with getting a "missing client" response from the BE, right? It was only when someone else would create a session with you with a prekey

File: Source/Model/UserClient/UserClient.swift [hunk 2]
```diff
@@ -464,15 +464,14 @@ public extension UserClient {
     
     /// Use this method only for the selfClient
     public func decrementNumberOfRemainingKeys() {
-        let oldValue = numberOfKeysRemaining
```
Comment (mikeger) [inline Source/Model/UserClient/UserClient.swift]: Sorry, I see that it was not there before, should we assert on calling this not on self client? 
`Use this method only for the selfClient`

File: Source/Model/UserClient/UserClient.swift (hunk 1 above)
Comment (typfel) [inline Source/Model/UserClient/UserClient.swift]: Yes, that's the only place where we are decrementing the number of prekeys

File: Source/Model/UserClient/UserClient.swift [hunk 3]
```diff
@@ -464,15 +464,14 @@ public extension UserClient {
...
-        if(numberOfKeysRemaining > 0) {
+        if numberOfKeysRemaining > 0 {
             numberOfKeysRemaining -= 1
         }
-        if(numberOfKeysRemaining < 0) { // this will recover from the fact that the number might already be < 0
-                                        // from a previous run
+        if numberOfKeysRemaining < 0 { // this will recover from the fact that the number might already be < 0
+                                       // from a previous run
             numberOfKeysRemaining = 0;
```
Comment (mikeger) [inline Source/Model/UserClient/UserClient.swift]: Would be cool to remove the semicolon (worst issue ever).

**Comment by typfel** [comment]
@mikeger updated

**Pull request merged**